crypto/src/utils.rs

```diff
@@ -60,63 +61,78 @@ pub fn read_keys_from_file<P: AsRef<Path>>(
     key_file.read_to_end(&mut file_content)?;
     let keys: EncryptedKeys =
         toml::from_slice(file_content.as_slice()).map_err(|e| Error::new(ErrorKind::Other, e))?;
-    let pub_key = PublicKey::from_slice(
-        &hex::decode(&keys.public_key).map_err(|e| Error::new(ErrorKind::Other, e))?,
-    ).ok_or_else(|| Error::new(
-        ErrorKind::Other,
-        "Couldn't create PublicKey from slice",
-    ))?;
-    let sec_key = SecretKey::from_slice(&decrypt(&keys, pass_phrase)?).ok_or_else(|| Error::new(
-        ErrorKind::Other,
-        "Couldn't create SecretKey from slice",
-    ))?;
+    let pub_key = keys.public_key;
+    let sec_key = keys.decrypt(pass_phrase)?;
```

Comment:
`EncryptedKeys` API can be improved.

It's weird to see the public key to be accessed via the `public_key` field when secret key is accessed via the `decrypt()` method. Using getter methods `public_key()` and `secret_key(passphrase: &[u8])` for both key types will be more consistent.

If you insist on consuming `EncryptedKeys` on decryption then `decrypt()` method may return a tuple:

```rust
let (pub_key, sec_key) = keys.decrypt(passphrase)?;
```

Reply:
Makes sense